Merge Win&lose conditions and skill systems.

IDokan opened the pull request:
It is a pull request that ask for merge functionality of skill system and win&lose system to the main branch.

IDokan committed:
Added textures for decals. Debugged access violations.

BP_Hero - Added a skill container on the BP.
BP_IKGameModeBase, BP_IKHUD, BP_IKPlayerController, BP_SkillBarWidget - A blueprints
HeroAITestLevel - A demo scene applied my codes to test them.
decal_albedo, decal_opacity - Textures for decal materials. (For sake of Visualizing Skill Ranges)
TargetingDecal - A new material for decal consists of a couple of textures.
MyTestSkill - Added a safe check before accessing a pointer.
SkillBarWidget - Removed a Log.
TargetingComponent - Added a safe check and tried adding decals.

IDokan committed:
#5 - Enable target visuals. Extend skill widgets.

BP_IKPlayerController - Add decal materials.
BP_SkillBarWidget - Added buttons.
HeroAITestLevel - Unknown changes..
IKPlayerController - Remove unnecessary codes.
SkillBarWidget - Added buttons, serialize hero that have skills and use it.
TargetingComponent - Properly added visual actors and components for targeting. Because PlayerController does not have actual body to visualize, I needed to add an actor for decals.

IDokan committed:
#5 - Project target location to a limit.

decal_albedo, decal_opacity - Added a small changes.
TargetingComponent - Add a function to project. Set visuals size as current_target_data_.

IDokan committed:
#15 - Created containers for only heroes.

IKGameModeBase - Created containers for only heroes.
SkillBarWidget - Use it appropriately on C++ codes.

IDokan committed:
#18 - Established Win&Lose conditions and events.

IKGameModeBase - As well as a hero container, populated a new container for enemies. Added a function that will remove an actor from each containers. Created events when win or lose. However, there are not enough implementation on call the functions when actors died. Need changes in BP_Hero, and BP_Enemy.

ChunMook-Kim reviewed (changes requested):
There are merge conflict between BP_Hero and AI Test scene level.

IDokan committed:
#19 - Recover legacy BP_Hero and level.

BP_Hero - Remove Skill Container component.
HeroAITestLevel - Set Gamemode to None.

ChunMook-Kim reviewed (changes requested):
Need to check there are no conflict.

ChunMook-Kim reviewed (approved):
Change approved.

IDokan committed:
Merge branch 'main' into Win&Lose
